nilearn/surface/surface.py

```diff
@@ -1829,14 +1830,30 @@ def get_data(img):
     img : :obj:`~surface.SurfaceImage` or :obj:`~surface.PolyData`
         SurfaceImage whose data to concatenate and extract.
 
+    ensure_finite : bool
+        If True, non-finite values such as (NaNs and infs) found in the
+        image will be replaced by zeros.
+
     Returns
     -------
     :obj:`~numpy.ndarray`
         Concatenated data across hemispheres.
     """
     if isinstance(img, SurfaceImage):
         data = img.data
-    return np.concatenate(list(data.parts.values()), axis=0)
+    data = np.concatenate(list(data.parts.values()), axis=0)
+
+    if ensure_finite:
+        non_finite_mask = np.logical_not(np.isfinite(data))
+        if non_finite_mask.sum() > 0:  # any non_finite_mask values?
+            warn(
+                "Non-finite values detected. "
+                "These values will be replaced with zeros.",
+                stacklevel=3,
+            )
+            data[non_finite_mask] = 0
```

Comment:
this constitutes 90% of the code of nilearn._utils.niimg.get_safe_data

https://github.com/nilearn/nilearn/blob/cd2871895581caacfcbd2bb443fa5d54304d94b5/nilearn/_utils/niimg.py#L59-L67

Reply:
consider refactoring